### PR TITLE
[INLONG-10881][SDK] Transform SQL support concat_ws function

### LINKED ISSUE #10881: [Feature][SDK] Transform SQL support Concat_ws function
### Description

- parent issue https://github.com/apache/inlong/issues/10796

CONCAT_WS(string1, string2, string3,…)
- Returns a string that concatenates STRING2, STRING3, … with a separator STRING1. The separator is added between the strings to be concatenated.
- Returns NULL If STRING1 is NULL. 
- Compared with concat(), concat_ws() automatically skips NULL arguments. 

### Use case

`concat_ws(’~’, ‘AA’, Null(STRING), ‘BB’, ‘’, ‘CC’)` ---> “AA~BB~~CC”

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by MOONSakura0614**
Fixes #10881 

### Modifications
Add a function class: ConcatWsFunction and its unit test.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ConcatWsFunction.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+import net.sf.jsqlparser.expression.NullValue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@TransformFunction(names = {"concat_ws"})
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ConcatWsFunction.java:32]: add descriptions

**Comment by luchunliang** [comment]
please fix the conflicting file

**Comment by luchunliang** [comment]
Please use independent UT class file to avoid UT conflicting file.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ConcatWsFunction.java (hunk 1 above)
Comment (MOONSakura0614) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ConcatWsFunction.java:38]: done

**Comment by MOONSakura0614** [comment]
> Please use independent UT class file to avoid UT conflicting file.

done and thank you for your suggestion~

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**